src/error-table.rkt

```diff
@@ -1,20 +1,365 @@
 #lang racket
 
-(require "points.rkt" "syntax/types.rkt" "core/localize.rkt" "timeline.rkt"
-         "common.rkt")
+(require racket/flonum racket/set)
+(require "points.rkt" "syntax/types.rkt" "core/localize.rkt" "common.rkt"
+         "ground-truth.rkt")
 
-(provide group-errors)
+(provide actual-errors predicted-errors)
 
-(define (group-errors expr pcontext)  
+(define (actual-errors expr pcontext)
   (match-define (cons subexprs pt-errorss)
     (flip-lists
      (hash->list (car (compute-local-errors (list expr) (*context*))))))
-
+  
   (define pt-worst-subexprs
-    (for/list ([pt-errors (in-list pt-errorss)])
-      (match-define (cons worst-subexpr pt-max-error)
-        (argmax cdr (map cons subexprs pt-errors)))
-      (if (> pt-max-error 2) worst-subexpr #f)))
+    (foldr append
```

Comment:
Would be much cleaner with reap/sow and a nested for loop.

Reply:
Hmm I am not able to figure out how reap/sow works...